src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/azuredevops/AzureDevOpsServerPullRequestDecorator.java

```diff
@@ -0,0 +1,407 @@
+
+            LOGGER.trace(String.format("Azure commentThreads count: %s ", azureCommentThreads.size()));
+            azureCommentThreads.removeIf(x -> x.getThreadContext() == null || x.isDeleted());
+            LOGGER.trace(String.format("Azure commentThreads AFTER REMOVE count: %s ", azureCommentThreads.size()));
+
+            for (PostAnalysisIssueVisitor.ComponentIssue issue : openIssues) {
+                String filePath = analysisDetails.getSCMPathForIssue(issue).orElse(null);
+                Integer line = issue.getIssue().getLine();
+                if (filePath != null) {
+                    try {
+                        filePath = "/" + filePath;
+                        LOGGER.trace(String.format("ISSUE: authorLogin: %s ", issue.getIssue().authorLogin()));
+                        LOGGER.trace(String.format("ISSUE: key: %s ", issue.getIssue().key()));
+                        LOGGER.trace(String.format("ISSUE: type: %s ", issue.getIssue().type().toString()));
+                        LOGGER.trace(String.format("ISSUE: severity: %s ", issue.getIssue().severity()));
+                        LOGGER.trace(String.format("ISSUE: componentKey: %s ", issue.getIssue().componentKey()));
+                        LOGGER.trace(String.format("ISSUE: getLocations: %s ", Objects.requireNonNull(issue.getIssue().getLocations()).toString()));
+                        LOGGER.trace(String.format("ISSUE: getRuleKey: %s ", issue.getIssue().getRuleKey()));
+                        LOGGER.trace(String.format("COMPONENT: getDescription: %s ", issue.getComponent().getDescription()));
+                        DbIssues.Locations locate = Objects.requireNonNull(issue.getIssue().getLocations());
+                        boolean isExitsThread = false;
+                        for (CommentThread azureThread : azureCommentThreads) {
+                            LOGGER.trace(String.format("azureFilePath: %s", azureThread.getThreadContext().getFilePath()));
+                            LOGGER.trace(String.format("filePath: %s (%s)", filePath, azureThread.getThreadContext().getFilePath().equals(filePath)));
+                            LOGGER.trace(String.format("azureLine: %d", azureThread.getThreadContext().getRightFileStart().getLine()));
+                            LOGGER.trace(String.format("line: %d (%s)", line, azureThread.getThreadContext().getRightFileStart().getLine() == locate.getTextRange().getEndLine()));
+
+                            if (azureThread.getThreadContext().getFilePath().equals(filePath)
+                                    && azureThread.getComments()
+                                    .stream()
+                                    .filter(c -> c.getContent().contains(issue.getIssue().key()))
+                                    .count() > 0 ) {
+
+                                if(issue.getIssue().getStatus() != Issue.STATUS_OPEN
+                                        && azureThread.getStatus() == CommentThreadStatus.ACTIVE) {
```

Comment:
We're nested a long way here: `try` -> `for` -> `if` -> `try` -> `for` -> `if` -> `if`. Is there any chance of breaking this method up into smaller units of work?

Reply:
Broke this up into smaller methods